packages/rum-recorder/src/domain/rrweb-snapshot/snapshot.ts

```diff
@@ -0,0 +1,589 @@
+  try {
+    const rules = s.rules || s.cssRules
+    return rules ? Array.from(rules).map(getCssRuleString).join('') : null
+  } catch (error) {
+    return null
+  }
+}
+
+function getCssRuleString(rule: CSSRule): string {
+  return isCSSImportRule(rule) ? getCssRulesString(rule.styleSheet) || '' : rule.cssText
+}
+
+function isCSSImportRule(rule: CSSRule): rule is CSSImportRule {
+  return 'styleSheet' in rule
+}
+
+function extractOrigin(url: string): string {
+  let origin
+  if (url.indexOf('//') > -1) {
+    origin = url.split('/').slice(0, 3).join('/')
+  } else {
+    origin = url.split('/')[0]
+  }
+  origin = origin.split('?')[0]
+  return origin
+}
+
+const URL_IN_CSS_REF = /url\((?:(')([^']*)'|(")([^"]*)"|([^)]*))\)/gm
+const RELATIVE_PATH = /^(?!www\.|(?:http|ftp)s?:\/\/|[A-Za-z]:\\|\/\/).*/
+const DATA_URI = /^(data:)([^,]*),(.*)/i
+export function absoluteToStylesheet(cssText: string | null, href: string): string {
+  return (cssText || '').replace(
+    URL_IN_CSS_REF,
+    (origin: string, quote1: string, path1: string, quote2: string, path2: string, path3: string) => {
+      const filePath = path1 || path2 || path3
+      const maybeQuote = quote1 || quote2 || ''
+      if (!filePath) {
+        return origin
+      }
+      if (!RELATIVE_PATH.test(filePath)) {
+        return `url(${maybeQuote}${filePath}${maybeQuote})`
+      }
+      if (DATA_URI.test(filePath)) {
+        return `url(${maybeQuote}${filePath}${maybeQuote})`
+      }
+      if (filePath[0] === '/') {
+        return `url(${maybeQuote}${extractOrigin(href)}${filePath}${maybeQuote})`
+      }
+      const stack = href.split('/')
+      const parts = filePath.split('/')
+      stack.pop()
+      for (const part of parts) {
+        if (part === '.') {
+          continue
+        } else if (part === '..') {
+          stack.pop()
+        } else {
+          stack.push(part)
+        }
+      }
+      return `url(${maybeQuote}${stack.join('/')}${maybeQuote})`
+    }
+  )
+}
+
+function getAbsoluteSrcsetString(doc: Document, attributeValue: string) {
+  if (attributeValue.trim() === '') {
+    return attributeValue
+  }
+
+  const srcsetValues = attributeValue.split(',')
+  // srcset attributes is defined as such:
+  // srcset = "url size,url1 size1"
+  const resultingSrcsetString = srcsetValues
+    .map((srcItem) => {
+      // removing all but middle spaces
+      const trimmedSrcItem = srcItem.trimLeft().trimRight()
+      const urlAndSize = trimmedSrcItem.split(' ')
+      // this means we have both 0:url and 1:size
+      if (urlAndSize.length === 2) {
+        const absUrl = absoluteToDoc(doc, urlAndSize[0])
+        return `${absUrl} ${urlAndSize[1]}`
+      } else if (urlAndSize.length === 1) {
+        const absUrl = absoluteToDoc(doc, urlAndSize[0])
+        return `${absUrl}`
+      }
+      return ''
+    })
+    .join(', ')
+
+  return resultingSrcsetString
+}
+
+export function absoluteToDoc(doc: Document, attributeValue: string): string {
```

Comment:
I am wondering if we could replace that by [normalizeUrl(url: string)](https://github.com/DataDog/browser-sdk/blob/master/packages/core/src/tools/urlPolyfill.ts#L3)